core/src/main/java/org/apache/gravitino/storage/relational/mapper/provider/base/SecurableObjectBaseSQLProvider.java

```diff
@@ -84,6 +89,78 @@ public String softDeleteRoleMetasByMetalakeId(@Param("metalakeId") Long metalake
         + " AND ro.deleted_at = 0) AND ob.deleted_at = 0";
   }
 
+  public String softDeleteObjectRelsByMetadataObject(
+      @Param("metadataObjectId") Long metadataObjectId,
+      @Param("metadataObjectType") String metadataObjectType) {
+    return "UPDATE "
+        + SECURABLE_OBJECT_TABLE_NAME
+        + " SET deleted_at = (UNIX_TIMESTAMP() * 1000.0)"
+        + " + EXTRACT(MICROSECOND FROM CURRENT_TIMESTAMP(3)) / 1000"
+        + " WHERE metadata_object_id = #{metadataObjectId} AND deleted_at = 0"
+        + " AND type = #{metadataObjectType}";
+  }
+
+  public String softDeleteObjectRelsByCatalogId(@Param("catalogId") Long catalogId) {
+    return "UPDATE "
+        + SECURABLE_OBJECT_TABLE_NAME
+        + " sect SET deleted_at = (UNIX_TIMESTAMP() * 1000.0)"
```

Comment:
Can we simply use this SQL as " 

```SQL
update SECURABLE_OBJECT_TABLE_NAME c set xxxxx

where sect.deleted_at = 0 and sec.type in ('catalog', 'schema'....) and exist (



.....
)
 
```

Reply:
`sec.type` different for sub-sentences.  `deleted_at`is ok.